src/main/java/seedu/rc4hdb/model/venues/Booking.java

```diff
@@ -0,0 +1,50 @@
+package seedu.rc4hdb.model.venues;
+
+import java.time.DayOfWeek;
+import java.time.LocalTime;
+
+import seedu.rc4hdb.model.resident.Resident;
+
+/**
+ * Represents a booking for a venue. Bookings cannot span multiple days.
+ */
+public abstract class Booking {
+    protected static int counter;
+
+    protected Resident resident;
+    protected LocalTime startTime;
+    protected LocalTime endTime;
+    protected DayOfWeek dayOfWeek;
+
+    protected int id;
+
+    /**
+     * Constructor for a Booking instance.
+     * @param resident The resident trying to book the venue.
+     * @param startTime The start time of the booking.
+     * @param endTime The end time of the booking.
+     */
+    public Booking(Resident resident, String startTime, String endTime) {
```

Comment:
Is it possible that users will input a `startTime`, and `endTime` that is nonsensical? If so, we could make an exception class for all venue booking purposes, and use it here.

Reply:
Thanks for pointing that out Neale. The original intention was to enforce these checks in a `Parser` class after `BookCommand` is implemented. That said, I certainly agree that having a `VenueException` or `BookingException` class will be useful. 😄